.github/workflows/documentation.yml

```
@@ -0,0 +1,27 @@
+name: Documentation
+
+on:
+  push:
+    branches:
+      - master
+    tags: '*'
+  pull_request:
+
+jobs:
+  build:
+    permissions:
+      contents: write
+      statuses: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: '1.6'
+      - name: Install dependencies
+        run: julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
+      - name: Build and deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # If authenticating with GitHub Actions token
+          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }} # If authenticating with SSH deploy key
+        run: julia --project=docs/ docs/make.jl
```

Why not using
```yaml
      - uses: julia-actions/julia-docdeploy@releases/v1
        env:
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```
?

---

I just used the best practices promoted by the Documenter.jl docs:
https://documenter.juliadocs.org/v1.0.1/man/hosting/#GitHub-Actions

If what you are writing is working as well (or better), I'd be happy to change it - could you in this case maybe make a proper suggestion such that I know exactly which code to replace?

---

I think that code block is quite outdated, given also the fact it's still referencing `actions/checkout@v2`.  I selected the lines to replace, 21-27

---

```suggestion
      - uses: julia-actions/julia-buildpkg@v1
      - uses: julia-actions/julia-docdeploy@v1
        env:
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```
I hope I got indentation right

---

Yep, LGTM. Thanks!

If you're up for it, maybe you can also create a PR to Documenter.jl 😬